include/exadg/compressible_navier_stokes/postprocessor/postprocessor.cpp

```diff
@@ -263,6 +268,25 @@ PostProcessor<dim, Number>::initialize_derived_fields()
     divergence.reinit();
   }
 
+  // shear rate
+  if(pp_data.output_data.write_shear_rate)
+  {
+    AssertThrow(pp_data.output_data.write_velocity == true,
+                dealii::ExcMessage("You need to activate write_velocity."));
```

Comment:
I am currently not sure whether we need this Assert, because in the  very lengthy if-statement above you already added `|| pp_data.output_data.write_shear_rate` (which should be the correct logic, i.e. we need to re-compute the velocity, but we do not have to write vtu-output for velocity in order to be able to write output for the shear rate). Would you like to check this? Of course, the same would then apply to previous implementations for `vorticity`, `divergence`, and potentially other quantities ...

Reply:
Yep those assertions can be removed safely, doing that for the shear rate, div(u) and vorticity. See screenshot, .pvtu file does not contain u, but the dependent quantities!
![Screenshot from 2023-02-17 11-52-37](https://user-images.githubusercontent.com/69867655/219627384-a76ab113-8d5b-4cc3-bb7e-66dc228461b7.png)

Reply:
thtanks for checking this